agroal-api/src/main/java/io/agroal/api/exceptionsorter/PostgreSQLExceptionSorter.java

```diff
@@ -17,6 +17,6 @@
     @Override
     @SuppressWarnings( "CallToSuspiciousStringMethod" )
     public boolean isFatal(SQLException e) {
-        return "08006".equals( e.getSQLState() );
+        return e.getSQLState().startsWith("08"); //08 is connection exceptions
```

Comment:
There are cases where SQLState may be null

```suggestion
	String sqlState = e.getSQLState();
        return sqlState != null && sqlState.startsWith("08"); //08 is connection exceptions
```